src/performanceMetric.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * @license
+ * This file is part of the Game Closure SDK.
+ *
+ * The Game Closure SDK is free software: you can redistribute it and/or modify
+ * it under the terms of the Mozilla Public License v. 2.0 as published by Mozilla.
+
+ * The Game Closure SDK is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Mozilla Public License v. 2.0 for more details.
+
+ * You should have received a copy of the Mozilla Public License v. 2.0
+ * along with the Game Closure SDK.  If not, see <http://mozilla.org/MPL/2.0/>.
+ */
+
+ /**
+ * @module performanceMetric
+ *
+ * Measures performance by keeping a window of delta times and providing a score using that window.
+ *
+**/
+
+var engine = null;
+var floor = Math.floor;
+
+var DEFAULT_RANK = 0;
+var DEFAULT_ALLOW_REDUCTION = true;
+
+	
+var PerformanceTester = Class(function () {
+	this.historySize = 200;
+	var _lastTick = Date.now();
+	var _history = [];
+	var historyIndex = 0;
+
+	this.init = function () {
+		this.LOWER_BOUND = 10;
+		this.HIGHER_BOUND = 60;
+		this.measuring = false;
+	};
+
+	this.startMeasuring = function() {
+		if (!this.measuring) {
+			this.measuring = true;
+			if (engine === null) {
+				import ui.Engine as Engine;
+				engine = Engine.get();
+			}
+			_lastTick = Date.now();
+			engine.subscribe('Tick', this, 'onTick');
+		}
+	};
+
+	this.stopMeasuring = function() {
+		if (this.measuring) {
+		  this.measuring = false;
+			engine.unsubscribe('Tick', this, 'onTick');
+		}
+	};
+
+	this.onTick = function(dt) {
+		var now = Date.now();
+		var delta = now - _lastTick;
+		_history[historyIndex++] = delta;
+		if (historyIndex >= this.historySize) {
+			historyIndex = 0;
+		}
+		this.getPerformanceScore();
+		_lastTick = now;
+	};
+
+	this.getAverageTicksSpeed = function() {
+		var result = 0;
+		for (var i = 0; i < _history.length; i++) {
+			result += _history[i];
+		}
+		result /= _history.length;
+		return result;
+	};
+
+	this.getPerformanceScore = function() {
+		var tickSpeed = this.getAverageTicksSpeed();
+		var ticksPerSecond = 1000 / tickSpeed;
+		var adjustedTicksPerSecond = Math.min(ticksPerSecond, 60);
+		var mappedScore = _map(adjustedTicksPerSecond, this.LOWER_BOUND, this.HIGHER_BOUND, 0, 100);
+		return Math.max(0, mappedScore);
+	}
+	
+	this.getParticleCount = function(count, performance) {
```

Comment:
Let's change this API to `this.getAdjustedParticleCount = function (count, performanceRank, allowReduction) {`